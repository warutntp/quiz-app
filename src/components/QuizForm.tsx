import React from "react";

interface QuizFormProps {
  name: string;
  setName: (name: string) => void;
  handleSubmitName: (e: React.FormEvent) => void;
}

const QuizForm: React.FC<QuizFormProps> = ({
  name,
  setName,
  handleSubmitName,
}) => {
  return (
    <form onSubmit={handleSubmitName} className="mb-4">
      <div className="mb-2">
        <label
          htmlFor="name"
          className="block text-sm font-medium text-gray-700"
        >
          Enter your name:
        </label>
        <input
          type="text"
          id="name"
          value={name}
          onChange={(e) => setName(e.target.value)}
          className="mt-1 p-2 block w-full border border-gray-300 rounded-md shadow-sm"
          required
        />
      </div>
      <button
        type="submit"
        className="w-full py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-700"
      >
        Start Quiz
      </button>
    </form>
  );
};

export default QuizForm;
