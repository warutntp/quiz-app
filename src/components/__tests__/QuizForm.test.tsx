// src/components/__tests__/QuizForm.test.tsx

import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import QuizForm from "../QuizForm";
import "@testing-library/jest-dom/extend-expect";

test("QuizForm", () => {
  const handleSubmitName = jest.fn();
  const setName = jest.fn();

  const { container } = render(
    <QuizForm name="" setName={setName} handleSubmitName={handleSubmitName} />
  );
  expect(container).toMatchSnapshot();
});

test("renders quiz form and submits correctly", () => {
  const handleSubmitName = jest.fn();
  const setName = jest.fn();
  render(
    <QuizForm name="" setName={setName} handleSubmitName={handleSubmitName} />
  );

  const input = screen.getByLabelText("Enter your name:");
  fireEvent.change(input, { target: { value: "John" } });
  expect(setName).toHaveBeenCalledWith("John");

  const button = screen.getByText("Start Quiz");
  fireEvent.click(button);
  expect(handleSubmitName).toHaveBeenCalled();
});
