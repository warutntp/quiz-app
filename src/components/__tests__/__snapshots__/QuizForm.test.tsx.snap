// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizForm 1`] = `
<div>
  <form
    class="mb-4"
  >
    <div
      class="mb-2"
    >
      <label
        class="block text-sm font-medium text-gray-700"
        for="name"
      >
        Enter your name:
      </label>
      <input
        class="mt-1 p-2 block w-full border border-gray-300 rounded-md shadow-sm"
        id="name"
        required=""
        type="text"
        value=""
      />
    </div>
    <button
      class="w-full py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-700"
      type="submit"
    >
      Start Quiz
    </button>
  </form>
</div>
`;
